### PR TITLE
chore(assert): release `assert@1.0.0-rc.1`

### PR DESCRIPTION
**Pull request by iuioiua**
Towards #4717

### CODE REVIEW

File: deno.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "graphviz": "npm:node-graphviz@^0.1.1",
 
     "@std/archive": "jsr:@std/archive@^0.224.0",
-    "@std/assert": "jsr:@std/assert@^0.225.3",
+    "@std/assert": "jsr:@std/assert@1.0.0-rc.1",
```
Comment (kt3k) [inline deno.json:16]: Thanks for updating this as well!

**Review by kt3k — APPROVED** [review]
Went through all APIs. All looked reasonable to me. LGTM.

**Comment by iuioiua** [comment]
I am being nitpicky, but I have some thoughts while reviewing this package:
1. Error messages end in periods ("."), when the `msg` argument is undefined. Why? This seems strange. Is this done in the Deno runtime or Web API?
2. `msg` arguments might be better named `msgSuffix`. WDYT?
3. Arguments could be better named. E.g., for `assertArrayIncludes()`, arguments could be named `array` and `subArray`, instead of `actual` and `expected`, respectively. WDYT?
4. I wonder whether extending `AssertionError` in some cases and providing more properties that add context to the error thrown is worthwhile. E.g., for `assertNotEquals()`:
    ```ts
    export class AssertNotEqualsError<T> extends AssertionError {
      actual: T;
      expected: T;

      constructor(actual: T, expected: T, msg?: string) {
        super(
          msg
            ? `Expected actual: ${actual} not to be: ${expected}: ${msg}`
            : `Expected actual: ${actual} not to be: ${expected}`,
        );
        this.name = this.constructor.name;
        this.actual = actual;
        this.expected = expected;
      }
    }
    ```
    This can be done in a backward-compatible manner while making the error thrown more ergonomic. WDYT?

**Comment by kt3k** [comment]
> Error messages end in periods ("."), when the msg argument is undefined. Why? This seems strange. Is this done in the Deno runtime or Web API?

Error messages ending with "." seem just common as well as the error message without "." (Try seach for "new Error" in std). This doesn't seem the issue for `std/assert` to me. (This should be discussed in broader scope)

> msg arguments might be better named msgSuffix. WDYT?

`msg` looks enough descriptive to me. `msgSuffix` looks rather confusing to me.

> Arguments could be better named. E.g., for assertArrayIncludes(), arguments could be named array and subArray, instead of actual and expected, respectively. WDYT?

`actual` and `expected` are conventional in this usage. These look just fine to me.

> I wonder whether extending AssertionError in some cases and providing more properties that add context to the error thrown is worthwhile. E.g., for assertNotEquals():

Sounds a good idea, but doesn't look like a blocker for 1.0.0-rc.1.

**Comment by iuioiua** [comment]
> Error messages ending with "." seem just common as well as the error message without "." (Try seach for "new Error" in std). This doesn't seem the issue for `std/assert` to me. (This should be discussed in broader scope)

There are 100+ `new Error`s in the Standard Library. 14 of them contain a period. So, periods in error messages are part of a minority. Either way, on second thought, let's not worry about this. It's inconsistent but inconsequential.

> `msg` looks enough descriptive to me. `msgSuffix` looks rather confusing to me.

It's appended to the error message. Either way, not blocking.

> `actual` and `expected` are conventional in this usage. These look just fine to me.

Yes, it's conventional, but it doesn't completely make sense. Either way, not blocking.

> Sounds a good idea, but doesn't look like a blocker for 1.0.0-rc.1. Sounds like a good enhancement candidate after the stabilization

I'll open a new issue when I can.

**Comment by iuioiua** [comment]
One last thing - should we consider #4934 a blocker?

**Comment by kt3k** [comment]
> One last thing - should we consider https://github.com/denoland/deno_std/pull/4934 a blocker?

#4934 is this issue

**Comment by kt3k** [comment]
If you mean #3771, then no. It can be landed as a fix after the stabilization.

**Comment by iuioiua** [comment]
My bad - typo. Ok.

**Pull request merged**